src/wp-includes/class-wp-token-map.php

```diff
@@ -0,0 +1,810 @@
+	 *     false === $smilies->contains( 'simile' );
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string  $word             Determine if this word is a lookup key in the map.
+	 * @param ?string $case_sensitivity 'case-insensitive' to ignore ASCII case or default of 'case-sensitive'.
+	 * @return bool Whether there's an entry for the given word in the map.
+	 */
+	public function contains( $word, $case_sensitivity = 'case-sensitive' ) {
+		$ignore_case = 'case-insensitive' === $case_sensitivity;
+
+		if ( $this->key_length >= strlen( $word ) ) {
+			if ( 0 === strlen( $this->small_words ) ) {
+				return false;
+			}
+
+			$term    = str_pad( $word, $this->key_length + 1, "\x00", STR_PAD_RIGHT );
+			$word_at = $ignore_case ? stripos( $this->small_words, $term ) : strpos( $this->small_words, $term );
+			if ( false === $word_at ) {
+				return false;
+			}
+
+			return true;
+		}
+
+		$group_key = substr( $word, 0, $this->key_length );
+		$group_at  = $ignore_case ? stripos( $this->groups, $group_key ) : strpos( $this->groups, $group_key );
+		if ( false === $group_at ) {
+			return false;
+		}
+		$group        = $this->large_words[ $group_at / ( $this->key_length + 1 ) ];
+		$group_length = strlen( $group );
+		$slug         = substr( $word, $this->key_length );
+		$length       = strlen( $slug );
+		$at           = 0;
+
+		while ( $at < $group_length ) {
+			$token_length   = unpack( 'C', $group[ $at++ ] )[1];
+			$token_at       = $at;
+			$at            += $token_length;
+			$mapping_length = unpack( 'C', $group[ $at++ ] )[1];
+			$mapping_at     = $at;
+
+			if ( $token_length === $length && 0 === substr_compare( $group, $slug, $token_at, $token_length, $ignore_case ) ) {
+				return true;
+			}
+
+			$at = $mapping_at + $mapping_length;
+		}
+
+		return false;
+	}
+
+	/**
+	 * If the text starting at a given offset is a lookup key in the map,
+	 * return the corresponding transformation from the map, else `false`.
+	 *
+	 * This function returns the translated string, but accepts an optional
+	 * parameter `$skip_bytes` which communicates how many bytes long the
+	 * lookup key was, if it found one. This can be used to advance a cursor
+	 * in calling code if a lookup key was found.
+	 *
+	 * Example:
+	 *
+	 *     false === $smilies->read_token( 'Not sure :?.', 0, $bytes_skipped );
+	 *     '😕'  === $smilies->read_token( 'Not sure :?.', 9, $bytes_skipped );
+	 *     2     === $bytes_skipped;
+	 *
+	 * Example:
+	 *
+	 *     while ( $at < strlen( $input ) ) {
+	 *         $next_at = strpos( $input, ':', $at );
+	 *         if ( false === $next_at ) {
+	 *             break;
+	 *         }
+	 *
+	 *         $smily = $smilies->read_token( $input, $next_at, $bytes_skipped );
+	 *         if ( false === $next_at ) {
+	 *             ++$at;
+	 *             continue;
+	 *         }
+	 *
+	 *         $prefix  = substr( $input, $at, $next_at - $at );
+	 *         $at     += $bytes_skipped;
+	 *         $output .= "{$prefix}{$smily}";
+	 *     }
```

Comment:
Note that this example uses `bytes_skipped` instead of `skip_bytes`, which is slightly confusing. It would be good to use the parameter name.

Reply:
they were different because of perspective. including `matched_` in the name I think helps the wording on the internal side, but I think that calling code still has a valid reason to use a different name. maybe I can further clarify with something like `$handle_length`